hivemind/optim/collaborative.py

```diff
@@ -263,12 +287,18 @@ def report_training_progress(self):
             self.should_report_progress.clear()
             with self.lock_local_progress:
                 current_time = get_dht_time()
-                local_state_info = [self.local_step, self.local_samples_accumulated,
-                                    self.performance_ema.samples_per_second, current_time, not self.averager.listen]
-
-            assert self.is_valid_peer_state(local_state_info), local_state_info
-            self.dht.store(self.training_progress_key, subkey=self.averager.endpoint, value=local_state_info,
-                           expiration_time=current_time + self.metadata_expiration, return_future=True)
+                local_state_info = TrainingState(
+                    endpoint=self.averager.endpoint,
+                    step=self.local_step,
+                    samples_accumulated=self.local_samples_accumulated,
+                    samples_per_second=self.performance_ema.samples_per_second,
+                    current_time=get_dht_time(),
```

Comment:
I meant that it's possible to use the value of the variable `current_time` in L289, since it already stores the recent value of `get_dht_time` :) `time` as a class field name was perfectly fine, it's probably best to keep it as is

Reply:
Got it, I've changed the names back :)